tests/integration/helpers/cluster.py

```diff
@@ -1834,6 +1856,11 @@ def add_instance(
                 self.setup_kafka_cmd(instance, env_variables, docker_compose_yml_dir)
             )
 
+        if with_kafka_sasl and not self.with_kafka_sasl:
+            cmds.append(
+                self.setup_kafka_sasl_cmd(instance, env_variables, docker_compose_yml_dir)
+            )
+
         if with_kerberized_kafka and not self.with_kerberized_kafka:
```

Comment:
I think in case of `with_kafka` and `with_kafka_sasl` both set to `True`, we might have some clashes, e.g.: both uses `self.kafka_port`. I am not sure if this causes any issues, but maybe it worth to check that the mentioned two options are not set at the same time.

Reply:
Can they ever be set at the same time? They're separate test suites. I was trying to keep it as much in line with the other tests. But I can completely decouple them by adding `kafka_port_sasl`, `kafka_sasl_docker_id`, `kafka_sasl_dir`?

Reply:
> Can they ever be set at the same time? 

Technically yes, but I am not sure what would be the usecase. Maybe we will has some issues when clickhouse connects to "normal" and "sasl" kafka instances at the same time.

It might not worth the effort, however if somebody sets them at the same time, let's prevent them shooting themselves in the foot. So I would say either add a check to make sure `with_kafka` and `with_kafka_sasl` are not set at the same time, or decouple them as you said. For me, both is perfectly fine.

Reply:
Ok, I've separated them. I'm not exactly sure why but now that the tests are separate `wait_kafka_is_available` no longer works. I'm not sure why it ever worked before since it's not using SASL credentials. It's probably possible to make it work with SASL - but it's also not needed since the test itself will retry setting up the `KafkaProducer`.